pallets/manta-pay/src/lib.rs

```diff
@@ -132,34 +142,33 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
 
-	/// Shards of the merkle tree of UTXOs
+	/// UTXOs and Encrypted Notes Grouped by Shard
 	#[pallet::storage]
 	pub(super) type Shards<T: Config> =
-		StorageDoubleMap<_, Identity, u8, Identity, u64, ([u8; 32], EncryptedNote), ValueQuery>;
+		StorageDoubleMap<_, Identity, u8, Identity, u64, (Utxo, EncryptedNote), ValueQuery>;
 
-	/// Shard merkle trees
+	/// Shard Merkle Tree Paths
 	#[pallet::storage]
 	pub(super) type ShardTrees<T: Config> =
 		StorageMap<_, Identity, u8, UtxoMerkleTreePath, ValueQuery>;
 
-	/// Outputs of Utxo accumulator
+	/// Outputs of Utxo Accumulator
 	#[pallet::storage]
 	pub(super) type UtxoAccumulatorOutputs<T: Config> =
-		StorageMap<_, Identity, [u8; 32], (), ValueQuery>;
+		StorageMap<_, Identity, UtxoAccumulatorOutput, (), ValueQuery>;
 
-	/// Utxo set of MantaPay protocol
+	/// UTXO Set
 	#[pallet::storage]
-	pub(super) type UtxoSet<T: Config> = StorageMap<_, Identity, [u8; 32], (), ValueQuery>;
+	pub(super) type UtxoSet<T: Config> = StorageMap<_, Identity, Utxo, (), ValueQuery>;
 
-	/// Void number set
+	/// Void Number Set
 	#[pallet::storage]
-	pub(super) type VoidNumberSet<T: Config> = StorageMap<_, Identity, [u8; 32], (), ValueQuery>;
+	pub(super) type VoidNumberSet<T: Config> = StorageMap<_, Identity, VoidNumber, (), ValueQuery>;
 
-	/// Void number set insertion order
-	/// Each element of the key is an `u64` insertion order number of void number.
+	/// Void Number Ordered by Insertion
 	#[pallet::storage]
-	pub(super) type VoidNumberSetInsertionOrder<T: Config> =
-		StorageMap<_, Identity, u64, [u8; 32], ValueQuery>;
+	pub(super) type InsertionOrderedVoidNumbers<T: Config> =
```

Comment:
this name change will need a migration on dolphin.
Unless we restart the testnet with this release.

Reply:
If this storage has been used in dolphin, we better do storage migration, and tx version might be bumped as well.